datafusion/core/src/datasource/physical_plan/json.rs

```diff
@@ -150,12 +151,22 @@ impl ExecutionPlan for NdJsonExec {
         target_partitions: usize,
         config: &datafusion_common::config::ConfigOptions,
     ) -> Result<Option<Arc<dyn ExecutionPlan>>> {
-        let repartition_file_min_size = config.optimizer.repartition_file_min_size;
+        let repartition_file_min_size =
+            if self.file_compression_type == FileCompressionType::GZIP {
+                OptimizerOptions::default().repartition_file_min_size
```

Comment:
I wonder it it would be simpler code to return `None` here
```suggestion
                None
```

That way the rest of the code could be left alone.

Reply:
Sure